src/battle/partner/sushie.c

```diff
@@ -1,19 +1,153 @@
 #include "common.h"
 
+extern f64 D_8023C060_70BD00;
+extern s32 D_8023C070;
+
+#ifdef NON_MATCHING
+ApiStatus func_80238000_707CA0(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* partnerActor = battleStatus->partnerActor;
+    ActorPart* partsTable;
+    f32 temp_f0;
+    f32 temp_f2;
+    f32 temp_f4;
+    f32 temp_f4_2;
+    f32 temp_f6;
+
+    set_goal_pos_to_part(partnerActor + 10, partnerActor->targetActorID, partnerActor->targetPartIndex);
+    partsTable = get_actor_part(get_actor(partnerActor->targetActorID), partnerActor->targetPartIndex);
+    temp_f2 = partnerActor->walk.goalPos.x + partsTable->unk_75;
+    temp_f6 = partnerActor->currentPos.x + 8;
+    temp_f4 = partnerActor->currentPos.y + 16;
+    partnerActor->walk.goalPos.x = temp_f2;
+    partnerActor->walk.currentPos.x = temp_f6;
+    partnerActor->walk.currentPos.y = temp_f4;
+    partnerActor->walk.goalPos.y = (f32) (partnerActor->walk.goalPos.y + (f32) partsTable->unk_76);
+    partnerActor->walk.currentPos.z = (f32) partnerActor->currentPos.z;
+    temp_f0 = atan2(temp_f6, temp_f4, temp_f2, partnerActor->walk.goalPos.y);
+    temp_f4_2 = (temp_f0 - 90.0f) * 0.25f;
+    partnerActor->walk.distance = temp_f0;
+    partnerActor->rotation.z = temp_f4_2;
+
+    if (temp_f4_2 < 0) {
+        partnerActor->rotation.z = 0;
+    }
+
+    return ApiStatus_DONE2;
+}
+#else
 INCLUDE_ASM(s32, "battle/partner/sushie", func_80238000_707CA0);
+#endif
 
 INCLUDE_ASM(s32, "battle/partner/sushie", func_80238114_707DB4);
 
-INCLUDE_ASM(s32, "battle/partner/sushie", func_802381EC_707E8C);
+ApiStatus func_802381EC_707E8C(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* partnerActor = battleStatus->partnerActor;
+    f32 xScale;
+    f32 yScale;
+    f32 zScale;
+    s32 temp_v0;
+
+    if (isInitialCall) {
+        script->functionTemp[0].s = 0;
+        script->functionTemp[1].s = 6;
+    }
+
+    xScale = partnerActor->scale.x;
+    yScale = partnerActor->scale.y;
+    zScale = partnerActor->scale.z;
+    partnerActor->scale.x = (f32) (xScale + ((1.0 - xScale) / 3.0));
```

Comment:
these f32 casts aren't needed (so you can get rid of it and the first layer of parens after)